src/utils/permissions.js

```diff
@@ -33,3 +21,20 @@ export function canUserEditVm (permissions: Array<PermissionType>): boolean {
 export function canUserUseAnyClusters (clusters: Array<ClusterType>): boolean {
   return clusters.find(cluster => cluster.get('canUserUseCluster')) !== undefined
 }
+
+export function getUserPermits (permissions: Array<PermissionType>): Array<string> {
+  const userFilter = Selectors.getFilter()
+  const userGroups = Selectors.getUserGroups()
+  const userId = Selectors.getUserId()
+  let permits = []
+  permissions.forEach((role) => {
+    if (userFilter ||
+      (
+        (role.groupId && userGroups.includes(role.groupId)) ||
+        (role.userId && role.userId === userId)
+      )) {
+      permits = [...permits, ...role.permits.map((permit) => permit.name)]
```

Comment:
Nit-pick: This will allow duplicate permits in the `permits` array if the user has multiple user/group/role mappings.  You could use a [ES6 Set](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) and then convert it to an array upon return.

Reply:
Good point.

Reply:
In this case I prefer to use Array instead of Set, because Set is too slow. I know that Set not so much slow, but still better to think about performance now, then when it will be to late.